### ISSUE TITLE
dotmap missing DC

### ISSUE THREAD

**Issue by sbma44** [openaddresses/openaddresses.io#77]
results.openaddresses.io shows a run as recently as 9/10 and the sample data and cache file look ok to me. It's present in the US South extract (haven't checked global--should I?). Not sure what could be going on.

![image](https://user-images.githubusercontent.com/31717/65274409-3bfe3600-daf1-11e9-8faa-39557d6c62a6.png)

**Comment by nvkelso** [comment]
I hit this today, too:

![image](https://user-images.githubusercontent.com/853051/66423039-afdc8180-e9bf-11e9-9562-fda538162fca.png)

![image](https://user-images.githubusercontent.com/853051/66423028-abb06400-e9bf-11e9-8740-5f386ffe7ad3.png)

**Comment by iandees** [comment]
Our style has four Mapbox studio layers ("southwest", "southeast", "northeast", "northwest") with the OA data split into each. When i look at one of these layers, the "select data" tab has an error e.g. "source layer 'openaddresses' or Source 'mapbox://open-addresses.southwest' not found.":

![image (1)](https://user-images.githubusercontent.com/261584/66423000-6183ae80-e9d0-11e9-932a-8c89a35611bd.png)

… but it's there:

![image (2)](https://user-images.githubusercontent.com/261584/66423053-77916f00-e9d0-11e9-8ab7-e82419eeb486.png)

When I try to select it, studio shows a different source getting selected at the top. (i selected southwest, but the source thing at the top shows southeast):

![image (3)](https://user-images.githubusercontent.com/261584/66423123-a576b380-e9d0-11e9-9528-6dcdf604f9c0.png)

**Comment by iandees** [comment]
@migurski do you remember why we wrote out chunks of mbtiles instead of one big one?

**Comment by migurski** [comment]
Yeah, we split the world into four to slide under a Mapbox limitation on upload size: https://github.com/openaddresses/machine/pull/631#issuecomment-299682129

**Comment by iandees** [comment]
@sbma44 or @ingalls, can you (a) check if Mapbox's limitation on file upload size is still present and (b) if there's anything we can do to fix the Mapbox Studio issue described above?

**Comment by ingalls** [comment]
@iandees Looks like the limit is 25gb per [Ref](https://docs.mapbox.com/help/troubleshooting/uploads/#accepted-file-types-and-transfer-limits)

What is the current size of the file, and are we uploading prebuild vector tiles, or raw geojson?

I'm not super familiar with the studio/tiles side of things but https://docs.mapbox.com/api/maps/#create-a-tileset looks like it could be a more permanent solution to our problem. However I image uploading individual tiles would take considerably longer than our current approach

**Comment by migurski** [comment]
We are uploading prebuilt MBTiles files full of vector tiles after running Tippecanoe in the build process.

**Comment by ingalls** [comment]
@migurski any chance you can pull the size of the current file? My guess is this is the issue once again :/

**Comment by iandees** [comment]
Looking at the logs, it seems that maybe the most recent two mbtiles builds on 9/29 and 10/1 stopped without finishing or uploading to Mapbox. I don't see any errors, so maybe the instance got terminated? It went from ~1600UTC to 0100UTC, which is ~8 hours. Maybe there's a time limit we need to bump up?

On the 9/11 build, most of the downloads failed because I changed permissions on the S3 bucket. The upload to Mapbox worked, though, so that's likely why there are so few points on the map as that's the most recent tileset in our account.

Actions to take:
- [x] Double-check machine's `openaddr-update-dotmap` is using the code that makes authenticated S3 requests instead of plain 'ol HTTP downloads _(it is)_
- [x] Increase the timeout on the job if there is one (although the slowness might come from trying to make requests through the CDN instead of using S3 directly) _(timeout isn't the problem)_

**Comment by iandees** [comment]
It looks like this is an issue with a hanging postgres query while pulling down the cached sources to use when iterating through all data.

This query is run and takes a *very* long time (times out after ~2 days):

```sql
SELECT MAX(id), source_path FROM runs
WHERE source_path IN (
    -- Get all source paths for successful runs in this set.
    SELECT source_path FROM runs
    WHERE set_id = 684369
  )
  -- Get only successful, merged runs.
  AND status = true
  AND (is_merged = true OR is_merged IS NULL)
GROUP BY source_path;
```

The Postgres analyzer says it should be relatively fast:

```
                                                     QUERY PLAN
--------------------------------------------------------------------------------------------------------------------
 HashAggregate  (cost=9914.83..9937.92 rows=2309 width=32)
   Group Key: runs.source_path
   ->  Nested Loop  (cost=859.73..8098.89 rows=363188 width=32)
         ->  HashAggregate  (cost=850.31..850.45 rows=14 width=28)
               Group Key: runs_1.source_path
               ->  Index Scan using runs_set_ids on runs runs_1  (cost=0.42..843.36 rows=2780 width=28)
                     Index Cond: (set_id = 684369)
         ->  Bitmap Heap Scan on runs  (cost=9.42..516.46 rows=129 width=32)
               Recheck Cond: ((source_path = runs_1.source_path) AND status AND (is_merged OR (is_merged IS NULL)))
               ->  Bitmap Index Scan on runs_source_path_idx  (cost=0.00..9.39 rows=129 width=0)
                     Index Cond: (source_path = runs_1.source_path)
(11 rows)
```

I'll dive into why this is taking so long later.

**Comment by iandees** [comment]
With all the changes above, I'm seeing more dots on the dotmap now. Including DC:

![image](https://user-images.githubusercontent.com/261584/67015164-b81d7680-f0bb-11e9-826f-8cc4eab351d4.png)

The update process takes quite a bit longer now (it's still running) that we have more data, but it's going somewhere now!
